### DISCUSSION TITLE
Export RSS on server.

### DISCUSSION THREAD

**Discussion by ghost** [FreshRSS/FreshRSS#5472] [Q&A] [answered]
Hi. I would like to automatically export RSS file from main feed, but not to the offline file but copy on the server and then copy then on my own domain on the same hosting. I tried it by the <code>wget</code> command  in cron, but I failed. Is it possible?

**Comment by Alkarex** [comment]
Hello,
If you define a token in your user settings, you can then access an RSS output of any FreshRSS view.

![image](https://github.com/FreshRSS/FreshRSS/assets/1008324/2212f504-25fd-478c-852c-fbfb3600ec51)

Likewise, you can also get an OPML of a view.

For server-side command line, check [our CLI ](https://github.com/FreshRSS/FreshRSS/blob/edge/cli/README.md), which offers ways to export/importing various things, including OPML.

**Comment by ghost** [comment]
Thanks.

I have open my default account so it's not a problem to show RSS for somebody. I was meant something else. To automation cron server to upload file from open access of default account of the instance and copy to somewhere else on the server. If it not possible nevermind.

By the way, I noticed that RSS exported from FreshRSS does not pass validation.
https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Frss.warroza.pl%2Fi%2F%3Fa%3Drss%26rid%3D6490688925800%26hours%3D168

**Reply by Alkarex** [reply]
> By the way, I noticed that RSS exported from FreshRSS does not pass validation. https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Frss.warroza.pl%2Fi%2F%3Fa%3Drss%26rid%3D6490688925800%26hours%3D168

Could you give a try to our current edge version (rolling release) of FreshRSS? I believe this might have been fixed already.

I have not fully understood what you try to do, but for automation, I suggest you look at [our CLI ](https://github.com/FreshRSS/FreshRSS/blob/edge/cli/README.md), which offers ways to export/importing various things, including OPML.

**Reply by hanyangl5** [reply]
I fixed that by adding a token, rsslink with empty token is invalid.

**Reply by Alkarex** [reply]
@hanyangl5 Please see new method https://github.com/FreshRSS/FreshRSS/discussions/5472#discussioncomment-8741981
While waiting for version 1.24, you can use it already now if you use our `edge` version

**Comment by Alkarex** [comment, marked as answer]
From FreshRSS 1.24, the recommended method to re-share RSS outputs is to use *user queries*, which allow sharing via RSS, HTML, and OPML:
https://freshrss.github.io/FreshRSS/en/users/user_queries.html#share-your-user-queries
The method with the *master token* is deprecated.
